src/editable-heading/editable-heading.tsx

```diff
@@ -121,15 +130,33 @@ export const EditableHeading = (props: EditableHeadingProps) => {
     setIsInSelectionMode(false);
   }, [isMouseDown, isInSelectionMode, disabled, onEdit]);
 
-  const onInputFocus = React.useCallback((e: React.FocusEvent<HTMLInputElement>) => {
-    setIsInFocus(true);
-    onFocus?.(e);
-  }, [onFocus]);
-
-  const onInputBlur = React.useCallback((e: React.FocusEvent<HTMLInputElement>) => {
-    setIsInFocus(false);
-    onBlur?.(e);
-  }, [onBlur]);
+  const onInputFocus = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+      setIsInFocus(true);
+      onFocus?.(e);
+    }, [onFocus]);
+
+  const onInputBlur = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+      setIsInFocus(false);
+      onBlur?.(e);
+    }, [onBlur]);
+
+  textAreaRef?.current?.addEventListener('scroll', () => {
+    const scrollHeight = textAreaRef?.current?.scrollHeight || 0;
+    const clientHeight = textAreaRef?.current?.clientHeight || 0;
+    const scrollTop = textAreaRef?.current?.scrollTop || 0;
+
+    const isScrolledToBottom = scrollHeight - clientHeight <= scrollTop;
+
+    if (isScrolledToBottom) {
```

Comment:
I can't understand why we have to remove `textareaWrapper` class when scrolled to bottom, maybe you can help me to understand?

Reply:
Yes, it's for fading in from the bottom of the textarea
<img width="726" alt="Screenshot 2023-11-04 at 18 59 04" src="https://github.com/JetBrains/ring-ui/assets/10141840/c120335d-687a-4053-8df7-07bf18e32ceb">

Reply:
Fade should only exist when we're not at the bottom of our scroll

Reply:
Here is the video to understand it better

Uploading Screen Recording 2023-11-04 at 19.06.16.mov…

Reply:
Here is the video to understand it better

Uploading Screen Recording 2023-11-04 at 19.06.16.mov…

Reply:
Okay, thank you! Maybe we could add more context to that classname by naming it like `...withShadow`, but that's does not matter too much